__tests__/app.test.js

```diff
@@ -0,0 +1,80 @@
+const app = require('../app.js');
+const data = require('../db/data/test-data/index.js');
+const seed = require('../db/seeds/seed.js');
+const db = require('../db/connection.js');
+const request = require('supertest');
+
+beforeEach(() => {
+    return seed(data);
+});
+
+afterAll(() => {
+    db.end();
+});
+
+describe('GET /api/topics', () => {
+    it('should return a 200 http status code', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200);
+    });
+
+    test('response data should be on a "topics" key', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body).toHaveProperty('topics');
+        });
+    });
+    
+    test('response data should contain three topics', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.topics).toHaveLength(3);
+        });
+    });
+
+    test('each topic should have correct object layout', () => {
+        const objectLayout = {
+            slug: expect.any(String),
+            description: expect.any(String)
+        };
+
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({ body }) => {
+            for (const topic of body.topics) {
+                expect(topic).toMatchObject(objectLayout);
+            }
+        });
+    });
+
+    test('response data should match the provided seed data', () => {
+        const seedTopics = data.topicData;
+
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.topics).toEqual(seedTopics);
+        });
+    });
```

Comment:
The above test is one I would usually avoid, in this case the table isn't providing an id for each topic so it passes. But you wouldn't be able to do this with articles or comments as they will only have id's after being inserted into the db - plus some data manipulation.